Helper/AlgoliaHelper.php

```diff
@@ -83,23 +77,25 @@ public function __construct(
 
     /**
      * @return void
-     * @throws AlgoliaException
      */
     protected function createClient(): void
     {
         $storeId = $this->getStoreId();
-        if ($this->algoliaCredentialsManager->checkCredentials($storeId)) {
-            $config = SearchConfig::create(
-                $this->config->getApplicationID($storeId),
-                $this->config->getAPIKey($storeId)
-            );
-            $config->setConnectTimeout($this->config->getConnectionTimeout($storeId));
-            $config->setReadTimeout($this->config->getReadTimeout($storeId));
-            $config->setWriteTimeout($this->config->getWriteTimeout($storeId));
-            $this->clients[$storeId] = SearchClient::createWithConfig($config);
-        } else {
-            throw new AlgoliaException('Client initialization could not be performed because Algolia credentials were not provided.');
```

Comment:
I agree we shouldn't be throwing an exception in the constructor but is there a reason we are eliminating it altogether?

Reply:
I thought it was better to reuse the method of the `AlgoliaCredentialsManager` class which handles both CLI and admin messages "gracefully". I don't mind changing it if you prefer.

Reply:
I hear ya - we are still using `\Algolia\AlgoliaSearch\Service\AlgoliaCredentialsManager::checkCredentials` which definitely has value - but this particular call impacts deeper in the code and I think the error message display should happen higher up the stack - for instance when we validate in `\Algolia\AlgoliaSearch\Model\Indexer\Category::execute`.